sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java

```diff
@@ -172,6 +176,10 @@ public ArrowColumnVector(ValueVector vector) {
       }
     } else if (vector instanceof NullVector) {
       accessor = new NullAccessor((NullVector) vector);
+    } else if (vector instanceof IntervalYearVector) {
+      accessor = new IntervalYearAccessor((IntervalYearVector) vector);
+    } else if (vector instanceof IntervalDayVector) {
+      accessor = new IntervalDayAccessor((IntervalDayVector) vector);
```

Comment:
Hm, there's something wrong here. We mapped `DayTimeIntervalType` to [`java.time.Duration`](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html) in Java but we map it here to `Interval` that represents a calendar instance (see also https://github.com/apache/arrow/blob/master/format/Schema.fbs).

I think we should map it to `DurationType` ([`datetime.timedelta`](https://docs.python.org/3/library/datetime.html#timedelta-objects)). I am working on SPARK-37277 to support this in Arrow conversion at PySpark but this became a blocker to me. I am preparing a PR to change this but please let me know if you guys have different thoughts.

Reply:
good catch!

Reply:
I'm not quite sure why `DayTimeIntervalType` map Arrow's `IntervalType` here. just according [ArrowUtils.scala#L60](https://github.com/apache/spark/blob/4ed2dab5ee46226f29919eba05fe52e446ab2449/sql/catalyst/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala#L60). I try to learn about [Arrow types](https://arrow.apache.org/docs/cpp/api/datatype.html). it's sql style. And in hive `INTERVAL_DAY_TIME` map arrow's `IntervalType` with `IntervalUnit.DAY_TIME` unit. If we map `DayTimeIntervalType` to  Arrow's `DurationType `. Then which type `YearMonthIntervalType` to match?

Reply:
At the very least `Duration` cannot be mapped to `YearMonthIntervalType`. Arrow-wise, mapping to `IntervalType` makes sense but it makes less sense in Spark SQL because we're already mapping different types.

I am not saying either way is 100% correct but I would pick the one to make it coherent in Spark's perspective if I have to pick one of both.

Reply:
and, `YearMonthIntervalType` is mapped to [`java.time.Period`](https://docs.oracle.com/javase/8/docs/api/java/time/Period.html) which is a calendar instance:

> A date-based amount of time in the ISO-8601 calendar system, such as '2 years, 3 months and 4 days'.

So `YearMonthIntervalType` seems fine.